pkg/reconciler/managed/reconciler.go

```diff
@@ -35,8 +35,9 @@ import (
 )
 
 const (
-	managedFinalizerName    = "finalizer.managedresource.crossplane.io"
-	managedReconcileTimeout = 1 * time.Minute
+	managedFinalizerName     = "finalizer.managedresource.crossplane.io"
+	reconcileTimeoutAddition = 30 * time.Second
```

Comment:
Grace period sounds better.

> Perhaps it should be configurable?

I'm not sure whether it should be configurable by provider stack developer. The commands that will be called in this period are against the api-server and that's not known about by the developer of the provider stack, i.e. external timeout is more relevant to them since that doesn't change, you always query the same entity but they don't know the quality of the k8s you're running their stack. Maybe it'd make sense to get it from an env var or flag to cover the case where api-server is so slow that the person who's running Crossplane wants to make it longer. Though I'm not sure whether it's worth adding a new flag.